MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt

```diff
@@ -158,11 +150,8 @@ object VisionManager : ARDataProvider {
     }
 
     // Event Listeners
-    private var visionEventsListener: VisionEventsListener? by Delegates.observable<VisionEventsListener?>(null) { _, _, new ->
-        if (::visionCore.isInitialized) {
-            visionCore.setVisionEventListener(new)
-        }
-    }
+    private var visionEventsListener: WeakReference<VisionEventsListener>? = null
```

Comment:
I've made them all wear references to align approaches.

Reply:
Ok. It is fine for me. But may be it is better to wrap listener to WeakReference at SDK level?